packages/esm-patient-clinical-view-app/src/store.ts

```diff
@@ -1,18 +1,35 @@
-import { useExtensionStore } from '@openmrs/esm-framework';
+import { useExtensionStore, useConfig } from '@openmrs/esm-framework';
 import startCase from 'lodash-es/startCase';
+import { useMemo } from 'react';
 
 export const useClinicalView = () => {
   const { slots } = useExtensionStore();
-  if (slots) {
-    const viewResults = Object.keys(slots)
-      .filter((slot) => slot.search('slot') !== -1)
-      .map((slot) => {
+  const { clinicalViews } = useConfig();
+  const viewResults = useMemo(
+    () =>
+      slots
+        ? Object.keys(slots)
+            .filter((slot) => slot.search('slot') !== -1)
+            .map((slot) => {
+              return {
+                slotName: startCase(slot.replace(/-/g, ' ').replace('slot', '')),
+                slot: slot,
```

Comment:
Wouldn't it be possible to just directly insert the `checked` attribute from below here (not 100% sure just from looking at the changes here on GitHub, but worth a try imo)? If it _does_ work, it makes the code a lot simpler here and removes the duplicate values (the `initialState` below currently holds two arrays with essentially the same objects, just that one has the `checked` attribute and the other one doesn't).